File: backend/src/main/java/turip/content/controller/dto/response/ContentResponse.java
package turip.content.controller.dto.response;

import java.time.LocalDate;
import turip.region.controller.dto.response.CityNameResponse;
import turip.content.domain.Content;
import turip.creator.controller.dto.response.CreatorResponse;

public record ContentResponse(
        Long id,
        String title,
        String url,
        LocalDate uploadedDate,
        CityNameResponse city,
        CreatorResponse creator,
        boolean isFavorite
) {
    public static ContentResponse of(Content content, boolean isFavorite) {
        return new ContentResponse(
                content.getId(),
                content.getTitle(),
                content.getUrl(),
                content.getUploadedDate(),
                CityNameResponse.from(content.getCity()),
                CreatorResponse.from(content.getCreator()),
                isFavorite
        );
    }
}

File: mobile/core/data/src/commonMain/kotlin/com/on/turip/core/data/dto/content/ContentResponse.kt
package com.on.turip.core.data.dto.content

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ContentResponse(
    @SerialName("creator")
    val creator: CreatorInformationResponse,
    @SerialName("id")
    val id: Long,
    @SerialName("title")
    val title: String,
    @SerialName("uploadedDate")
    val uploadedDate: String,
    @SerialName("url")
    val url: String,
    @SerialName("city")
    val city: CityResponse,
    @SerialName("isBookmarked")
    val isBookmarked: Boolean,
)
